src/Transports.AspNetCore/WebSockets/GraphQLWs/SubscriptionServer.cs

```diff
@@ -0,0 +1,261 @@
+    /// <param name="serializer">The <see cref="IGraphQLSerializer"/> to use to deserialize payloads stored within <see cref="OperationMessage.Payload"/>.</param>
+    /// <param name="serviceScopeFactory">A <see cref="IServiceScopeFactory"/> to create service scopes for execution of GraphQL requests.</param>
+    /// <param name="userContextBuilder">The user context builder used during connection initialization.</param>
+    /// <param name="authenticationService">An optional service to authenticate connections.</param>
+    public SubscriptionServer(
+        IWebSocketConnection sendStream,
+        GraphQLWebSocketOptions options,
+        IAuthorizationOptions authorizationOptions,
+        IDocumentExecuter executer,
+        IGraphQLSerializer serializer,
+        IServiceScopeFactory serviceScopeFactory,
+        IUserContextBuilder userContextBuilder,
+        IWebSocketAuthenticationService? authenticationService = null)
+        : base(sendStream, options, authorizationOptions)
+    {
+        DocumentExecuter = executer ?? throw new ArgumentNullException(nameof(executer));
+        ServiceScopeFactory = serviceScopeFactory ?? throw new ArgumentNullException(nameof(serviceScopeFactory));
+        UserContextBuilder = userContextBuilder ?? throw new ArgumentNullException(nameof(userContextBuilder));
+        Serializer = serializer ?? throw new ArgumentNullException(nameof(serializer));
+        _authenticationService = authenticationService;
+    }
+
+    /// <inheritdoc/>
+    public override async Task OnMessageReceivedAsync(OperationMessage message)
+    {
+        if (message.Type == MessageType.Ping)
+        {
+            await OnPingAsync(message);
+            return;
+        }
+        else if (message.Type == MessageType.Pong)
+        {
+            await OnPongAsync(message);
+            return;
+        }
+        else if (message.Type == MessageType.ConnectionInit)
+        {
+            if (Initialized)
+            {
+                await ErrorTooManyInitializationRequestsAsync(message);
+            }
+            else
+            {
+                await OnConnectionInitAsync(message, true);
+            }
+            return;
+        }
+        if (!Initialized)
+        {
+            await ErrorNotInitializedAsync(message);
+            return;
+        }
+        switch (message.Type)
+        {
+            case MessageType.Subscribe:
+                await OnSubscribeAsync(message);
+                break;
+            case MessageType.Complete:
+                await OnCompleteAsync(message);
+                break;
+            default:
+                await ErrorUnrecognizedMessageAsync(message);
+                break;
+        }
+    }
+
+    /// <summary>
+    /// GQL_PONG is a requrired response to a ping, and also a unidirectional keep-alive packet,
+    /// whereas GQL_PING is a bidirectional keep-alive packet.
```

Comment:
```suggestion
    /// Pong is a required response to a ping, and also a unidirectional keep-alive packet,
    /// whereas ping is a bidirectional keep-alive packet.
```

Only SubscriptionsTransportWs namespace has GQL prefixes.